doc/source/user_guide/io.rst

```diff
@@ -1806,6 +1807,22 @@ The ``Series`` object also has a ``to_string`` method, but with only the ``buf``
 ``na_rep``, ``float_format`` arguments. There is also a ``length`` argument
 which, if set to ``True``, will additionally output the length of the Series.
 
+.. _formatting missing values:
+
+   **Formatting missing values**
+
+   Formatting functions passed to the ``formatters`` argument above, are not
+   called for *missing values*, *string data types*, or ``PandasObject``.
```

Comment:
Apologies, I think I was thrown off by `StringDType` [here](https://github.com/pandas-dev/pandas/blob/d831326e11971af61307e4fc940855556ef3cdb7/pandas/io/formats/format.py#L1214), thinking it was an abstraction for multiple string types.

Am I wrong in the same way for `PandasObject`?

Looking at that section again, I'm finding the logic for when `na` values are _not_ rendered using the value specified in `na_rep`, rather baffling:
- `None` is treated differently to `na`. Sure, that probably indicates a storage issue, not a missing value at the source - but in that case shouldn't it raise an exception or in some other way flag an error somewhere?
- The constant `NA` appears to have a different meaning than what `isna()` checks for, with the string representation of the `NA` constant used if the value to be formatted _'satisfies'_ `isna()`, but is **also** equal to the _literal_ value of the `NA` constant. If `NA` _wasn't_ a constant I'd think it may hold the user-specified value equal to how missing values are represented in the _source_ data. But even if that was the case, returning the literal value of the `NA` variable would sidestep what `na_rep` is supposed to be doing...